### PR TITLE
`AddPropery` now accepts a comment

### PR DESCRIPTION
**Pull request by ammachado**
## What's changed?

`AddSpringProperty` from `rewrite-spring` accepts comments, but only applies it's a YAML file. This PR changes `AddProperty` to accept comments.

## What's your motivation?

Add properties to Spring applications.

### Checklist
- [x] I've added unit tests to cover both positive and negative cases
- [x] I've read and applied the [recipe conventions and best practices](https://docs.openrewrite.org/authoring-recipes/recipe-conventions-and-best-practices)
- [x] I've used the IntelliJ IDEA auto-formatter on affected files

### CODE REVIEW

File: rewrite-properties/src/main/java/org/openrewrite/properties/AddProperty.java [hunk 1]
```diff
@@ -41,13 +44,37 @@ public class AddProperty extends Recipe {
...
+    /**
+     * Keeping this constructor just for compatibility purposes
+     * @deprecated Use {@link AddProperty#AddProperty(String, String, String, String)}}
+     */
+    @Deprecated
+    public AddProperty(String property, String value, @Nullable String delimiter) {
+        this(property, value, null, delimiter);
+    }
+
+    @JsonCreator
```
Comment (timtebeek) [inline rewrite-properties/src/main/java/org/openrewrite/properties/AddProperty.java:70]: I could even lean towards removing this old constructor and making the change downstream if needed (which I'm open to do). That way there's no confusion going forward, especially as the three and four argument constructors have a different third argument, which could be confusing.
```suggestion
```

File: rewrite-properties/src/main/java/org/openrewrite/properties/AddProperty.java [hunk 2]
```diff
@@ -15,6 +15,7 @@
  */
 package org.openrewrite.properties;
 
+import com.fasterxml.jackson.annotation.JsonCreator;
```
Comment (timtebeek) [inline rewrite-properties/src/main/java/org/openrewrite/properties/AddProperty.java:18]: Could be removed if we delete the backwards compatible constructor.
```suggestion
```

**Review by timtebeek — APPROVED** [review]
Nice addition, thanks! Approved already; let me know how you'd feel about not having a backwards compatible constructor.

**Comment by ammachado** [comment]
> Nice addition, thanks! Approved already; let me know how you'd feel about not having a backwards compatible constructor.

It makes sense. I removed the constructor.

**Pull request merged**